ddspipe_participants/src/cpp/writer/rtps/CommonWriter.cpp

```diff
@@ -180,24 +190,20 @@ utils::ReturnCode CommonWriter::write_nts_(
         return ret;
     }
 
+    if (rtps_history_->isFull())
```

Comment:
```suggestion
    if (topic_.topic_qos.is_reliable() && rtps_history_->isFull())
```

Reply:
This would prevent taking a mutex in some cases (add this comment for developers).

Reply:
The function isFull doesn't take a mutex.